litho-widget/src/main/java/com/facebook/litho/widget/LinearLayoutInfo.java

```diff
@@ -134,19 +128,34 @@ public int getChildWidthSpec(int widthSpec, RenderInfo renderInfo) {
     }
   }
 
-  private static class LinearLayoutManagerNoPredictiveAnim
+  private static class InternalLinearLayoutManager
       extends LinearLayoutManager {
 
-    public LinearLayoutManagerNoPredictiveAnim(
+    private final boolean supportsPredictiveItemAnimations;
+
+    InternalLinearLayoutManager(
         Context context,
         int orientation,
         boolean reverseLayout) {
       super(context, orientation, reverseLayout);
+      // Predictive animation in RecyclerView has some bugs that surface occasionally when one
+      // RecyclerView is used as a child of another RecyclerView and it is very hard to repro
+      // (https://issuetracker.google.com/issues/37007605). This one disables that optional
+      // feature for Horizontal one which is used as inner RecyclerView most of the cases.
+      // Disabling predictive animations lets RecyclerView use the simple animations instead.
+      supportsPredictiveItemAnimations = (orientation == OrientationHelper.VERTICAL);
+    }
+
+    @Override
+    public LayoutParams generateDefaultLayoutParams() {
+      return getOrientation() == OrientationHelper.VERTICAL
+          ? new RecyclerView.LayoutParams(MATCH_PARENT, WRAP_CONTENT)
+          : new RecyclerView.LayoutParams(WRAP_CONTENT, MATCH_PARENT);
     }
 
     @Override
     public boolean supportsPredictiveItemAnimations() {
-      return false;
+      return supportsPredictiveItemAnimations;
```

Comment:
We only need to handle the force disabling case, and delegate the opposite case to parent as there is some logic involved in `LinearLayoutManager#supportsPredictiveItemAnimations()`. How about inverting the variable to something like `forceDisablePredictiveAnimations` and replace this line with:

`return forceDisablePredictiveAnimations ? false : super. supportsPredictiveItemAnimations()`

Reply:
Good catch, I didn't see the logic in the super call. Updated.